libraries/oakestra_utils_library/oakestra_utils/types/statuses.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional
+
+from oakestra_utils.types.common import CustomEnum
+
+# Legend
+# - 🟨: Only mentioned in Schedulers.
+# - 🟦: Explicitly mentioned in the Go NodeEngine and in python code.
+# - 🟪: Only mentioned in the Go NodeEngine.
+# - All others are only explicitly mentioned in non-scheduler Python code.
+
+
+class Status(CustomEnum):
+    pass
+
+
+class LegacyStatus(Status):
+    """TODO: Investigate how these legacy statuses are used and how to replace them gracefully.
+
+    Note: These values had been ints before"""
+
+    LEGACY_0 = "0"
+    LEGACY_1 = "1"
+
+
+class SchedulingStatus(Status):
+    "Represents the state before a service gets deployed"
+    pass
+
+
+class NegativeSchedulingStatus(SchedulingStatus):
+    TARGET_CLUSTER_NOT_FOUND = "TargetClusterNotFound"  # 🟨
+    TARGET_CLUSTER_NOT_ACTIVE = "TargetClusterNotActive"  # 🟨
+    TARGET_CLUSTER_NO_CAPACITY = "TargetClusterNoCapacity"  # 🟨
+
+    NO_ACTIVE_CLUSTER_WITH_CAPACITY = "NoActiveClusterWithCapacity"  # 🟨
+
+    NO_WORKER_CAPACITY = "NO_WORKER_CAPACITY"
+    NO_QUALIFIED_WORKER_FOUND = "NO_QUALIFIED_WORKER_FOUND"  # 🟨
+    NO_NODE_FOUND = "NO_NODE_FOUND"  # 🟨
+
+
+class PositiveSchedulingStatus(SchedulingStatus):
+    REQUESTED = "REQUESTED"
+    CLUSTER_SCHEDULED = "CLUSTER_SCHEDULED"
+    NODE_SCHEDULED = "NODE_SCHEDULED"
+
+
+class DeploymentStatus(Status):
+    """Represents the state of a service during deployment,
+    after it got scheduled."""
+
+    CREATING = "CREATING"  # 🟪
+
+    RUNNING = "RUNNING"
+    # TODO: Figure out strong difference between ACTIVE and RUNNING states.
+    # If there are no meaningful difference - merge both states into one.
+    # Else - adjust their names to emphasize the differences.
+    ACTIVE = "ACTIVE"  # 🟪
+
+    COMPLETED = "COMPLETED"  # 🟪
+
+    FAILED = "FAILED"  # 🟦 # TODO: Investigate the difference to DEAD
```

Comment:
In theory, but as we know theory and practice don't always match:

FAILED: exit status != 0
DEAD: exist status == 0

Reply:
Nice clarification. I added explanatory comments.